src/service/resources-impl.js

```diff
@@ -239,43 +279,157 @@ export class ResourcesImpl {
     }
   }
 
+  /** @override */
+  isExperimentOn(name) {
+    if (name == 'intersect-resources') {
+      return !!this.intersectionObserver_;
+    }
+    return false;
+  }
+
+  /**
+   * @param {!Array<!IntersectionObserverEntry>} entries
+   * @param {!IntersectionObserver} unusedObserver
+   * @private
+   */
+  intersects_(entries, unusedObserver) {
+    dev().fine(TAG_, 'intersect', entries);
+
+    const toUnload = [];
+
+    const promises = entries.map(entry => {
+      const {boundingClientRect, isIntersecting, target, rootBounds} = entry;
+      // Closure Compiler doesn't recognize boundingClientRect as a ClientRect.
+      const clientRect = /** @type {!ClientRect} */ (boundingClientRect);
+      devAssert(target.isUpgraded());
```

Comment:
Is this a safe assertion?

Reply:
Yea, we only start observing on upgrade.